Tests/Tactics/CSE.lean

```diff
@@ -269,80 +269,81 @@ hx2 : x9 + x3 = x2
 x28 : BitVec 64
 hx27 : BitVec.zeroExtend 128 x28 = x27
 x29 : BitVec 64
-hx26 : BitVec.zeroExtend 128 x29 = x26
+hx20 : x29 ^^^ x25 = x20
 x30 : BitVec 64
-hx20 : x30 ^^^ x25 = x20
-x31 : BitVec 64
-hx21 : x24 ^^^ x31 = x21
-x32 x33 : BitVec 64
-hx24 : x33 ^^^ x32 = x24
-x34 : BitVec 64
-x35 : BitVec (63 - 0 + 1)
-hx35 : BitVec.extractLsb 63 0 d = x35
+hx26 : BitVec.zeroExtend 128 x30 = x26
+x31 x32 : BitVec 64
+hx23 : x31 ^^^ x32 = x23
+x33 x34 : BitVec 64
+hx21 : x23 ^^^ x34 = x21
+x35 : BitVec (127 - 64 + 1)
+hx35 : BitVec.extractLsb 127 64 d = x35
+hx7 : x8 + x35 = x7
 x36 : BitVec (63 - 0 + 1)
-hx36 : BitVec.extractLsb 63 0 c = x36
-hx28 : x36 + x35 = x28
-x37 : BitVec (127 - 64 + 1)
-hx37 : BitVec.extractLsb 127 64 c = x37
-hx19 : x37 + x21 = x19
-x38 : BitVec (63 - 0 + 1)
-hx38 : BitVec.extractLsb 63 0 a = x38
-x39 : BitVec (63 - 0 + 1)
-hx39 : BitVec.extractLsb 63 0 b = x39
-hx1 : x2 + x39 = x1
-hx5 : x39 + x6 = x5
-x40 : BitVec (127 - 64 + 1)
-hx40 : BitVec.extractLsb 127 64 b = x40
-hx31 : x40.rotateRight 41 = x31
-hx32 : x40.rotateRight 18 = x32
-hx33 : x40.rotateRight 14 = x33
-hx34 : ~~~x40 = x34
-hx30 : x40 &&& x38 = x30
-x41 : BitVec (127 - 64 + 1)
-hx41 : BitVec.extractLsb 127 64 a = x41
-hx25 : x34 &&& x41 = x25
+hx36 : BitVec.extractLsb 63 0 a = x36
+x37 : BitVec (63 - 0 + 1)
+hx37 : BitVec.extractLsb 63 0 c = x37
+x38 : BitVec (127 - 64 + 1)
+hx38 : BitVec.extractLsb 127 64 c = x38
+hx19 : x38 + x21 = x19
+hx28 : x38 + x35 = x28
+x39 : BitVec (127 - 64 + 1)
+hx39 : BitVec.extractLsb 127 64 e = x39
+hx6 : x7 + x39 = x6
+hx15 : x17 + x39 = x15
+x40 : BitVec (63 - 0 + 1)
+hx40 : BitVec.extractLsb 63 0 b = x40
+hx1 : x2 + x40 = x1
+hx5 : x40 + x6 = x5
+x41 : BitVec (63 - 0 + 1)
+hx41 : BitVec.extractLsb 63 0 d = x41
+hx30 : x37 + x41 = x30
 x42 : BitVec (127 - 64 + 1)
-hx42 : BitVec.extractLsb 127 64 e = x42
-hx6 : x7 + x42 = x6
-hx15 : x16 + x42 = x15
+hx42 : BitVec.extractLsb 127 64 a = x42
+hx25 : x33 &&& x42 = x25
 x43 : BitVec (127 - 64 + 1)
-hx43 : BitVec.extractLsb 127 64 d = x43
-hx7 : x8 + x43 = x7
-hx29 : x37 + x43 = x29
+hx43 : BitVec.extractLsb 127 64 b = x43
+hx31 : x43.rotateRight 14 = x31
+hx32 : x43.rotateRight 18 = x32
+hx33 : ~~~x43 = x33
+hx34 : x43.rotateRight 41 = x34
+hx29 : x43 &&& x36 = x29
 x44 : BitVec (63 - 0 + 1)
 hx44 : BitVec.extractLsb 63 0 e = x44
-hx14 : x17 + x44 = x14
+hx14 : x16 + x44 = x14
 ⊢ x2 ++
-      ((x1 &&& x40 ^^^ ~~~x1 &&& x38) + (x1.rotateRight 14 ^^^ x1.rotateRight 18 ^^^ x1.rotateRight 41) +
+      ((x1 &&& x43 ^^^ ~~~x1 &&& x36) + (x1.rotateRight 14 ^^^ x1.rotateRight 18 ^^^ x1.rotateRight 41) +
         BitVec.extractLsb 63 0 x4) =
     x6 ++
-      (x36 + (x5.rotateRight 14 ^^^ x5.rotateRight 18 ^^^ x5.rotateRight 41) + (x5 &&& x40 ^^^ ~~~x5 &&& x38) + x35 +
+      (x37 + (x5.rotateRight 14 ^^^ x5.rotateRight 18 ^^^ x5.rotateRight 41) + (x5 &&& x43 ^^^ ~~~x5 &&& x36) + x41 +
         x44)
 -/
 #guard_msgs in theorem sha512h_rule_1 (a b c d e : BitVec 128) :
```

Comment:
I wonder if this subgoal guard_msgs needs to be recorded. It seems very fragile.

Reply:
@pennyannn Yeah, I'm not particularly happy about the state of affairs either :( I'm keeping it around in hopes that we will eventually get regex matching on `guard_msgs`.

Reply:
Yes, I'm also expecting this to be stopgap, one way or the other :-) Maybe `cse` can elide var names when printing, or some such.